EightMinuteEmpireLegends/Core/Map/Map.cpp

```diff
@@ -245,11 +292,25 @@ bool Vertex::compare(Vertex* v) {
 	return this->id == v->getId();
 }
 
+Vertex& Vertex::operator =(const Vertex& v) {
```

Comment:
What about the other attributes? I think for the assignment operator you must make it similar to the copy constructor, thus we need to make a deep copy.